File: Go/day02/day-02-a.go
package main

import (
	"fmt"
	"os"
	"strings"
)

func main() {
	bytes, _ := os.ReadFile("./input.txt");
	text := strings.TrimSpace(string(bytes))
	
	pointMap := map[byte]int{
		'A': 1,
		'X': 1,
		'B': 2,
		'Y': 2,
		'C': 3,
		'Z': 3,
	}

	point := 0

	for _, round := range strings.Split(text, "\n") {
		opponent, me := pointMap[round[0]], pointMap[round[2]]

		point += calculatePoint(opponent, me)
	}

	fmt.Println(point)
}


func calculatePoint(opponent, me int ) int {
	point := 0
	diff := abs(opponent,me)

	if opponent == me {
		point += 3
	} else if (diff ==1 && me > opponent) || (me == 1 && opponent == 3) {
		point += 6
	}
	point += me
	return point
}


func abs(a,b int) int {
	diff := a-b

	if diff < 0 {
		diff *= -1
	}

	return diff
}

File: Javascript/day02/day-02-a.js
const fs = require("fs");

const text = fs.readFileSync("./input.txt", "utf8").trimEnd();

const pointMap = new Map([
  ["A", 1],
  ["X", 1],
  ["B", 2],
  ["Y", 2],
  ["C", 3],
  ["Z", 3],
]);

let point = 0;

text.split("\n").forEach((round) => {
  let [opponent, me] = [pointMap.get(round[0]), pointMap.get(round[2])];
  point += calculatePoint(opponent, me);
});

function calculatePoint(opponent, me) {
  let point = 0;
  let diff = Math.abs(opponent - me);

  if (opponent == me) {
    point += 3;
  } else if ((diff == 1 && me > opponent) || (me == 1 && opponent == 3)) {
    point += 6;
  }
  point += me;

  return point;
}

console.log(point);
